compiler/optimizations/gpuTransforms.cpp

```diff
@@ -1615,6 +1616,34 @@ static void cleanupTaskIndependentCapturePrimitives() {
       cleanupTaskIndependentCapturePrimitive(callExpr);
 }
 
+static void reportErrorsForBadBlockSizeCalls() {
+  CallExpr* explainAnchor = nullptr;
+  for_alive_in_Vec(CallExpr, callExpr, gCallExprs) {
+    if(callExpr->isPrimitive(PRIM_GPU_SET_BLOCKSIZE)) {
+      USR_FATAL_CONT(callExpr, "'setBlockSize' must only be used in bodies of GPU-eligible loops");
+      explainAnchor = callExpr;
+
+      debuggerBreakHere();
+
+      // Search forward to try find the CForLoop corresponding to the GPU loop.
+      auto search = callExpr->next;
+      while (search) {
+        auto condStmt = toCondStmt(search);
+        if (gpuBranches.count(condStmt) > 0) {
```

Comment:
Just so I am following the logic here: we are finding the next conditional from the offending call with the assumption that it is the dynamic check we introduce per kernel launch site?

Reply:
Yep! This is a little heuristic: if we match the pattern, we can print a nicer error message in cases like the original motiviating case:

```Chapel
setBlockSize(128);
foreach 1..100 { ... }
```

Reply:
Understood. Could you update the comment to refer to the fact that we are actually looking for a conditional in the implementation and what that conditional is?